### DISCUSSION TITLE
ReadTimeout

### DISCUSSION THREAD

**Discussion by YupChutLuke** [deepgram/community#716] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Pre-recorded

### Details

I am working with the speech-to-text, pre-recorded endpoint. While working I have started to run into, about 50-75% of the time, an https.ReadTimeout error. 

Now the audio file is being pulled from s3 and then wrapped in a BytesIO object and I am using the transcribe_file method. Right now I am attempting to benchmark the transcribe_url vs. transcribe_file. However, on both I am hitting this timeout. The audio file is only 66 seconds long and I wouldn't imagine that this should be happening. Here is a code snippet.

```
def get_s3_file(key):
    s3 = boto3.client('s3')
    obj = s3.get_object(Bucket='vixzy-web', Key=key)
    data = obj['Body'].read()

    return BytesIO(data)
def main():
    key = <key_name>

    io_obj = get_s3_file(key) #
    API_KEY = os.getenviro("API_KEY")
    deepgram = DeepgramClient(API_KEY)

    options = PrerecordedOptions(
        model="nova-2-phonecall",
        smart_format=True,
        profanity_filter=True,
        redact='pci',
        language='en',
        diarize=True,
        topics=True,
        custom_topics=[<Topic>],
        custom_topic_mode="extended",
        intents=True,
        custom_intent=[<INTENTS>],
        custom_intent_mode="extended"
    )

 
    payload: FileSource = {
        "buffer": io_obj
    }
    response = deepgram.listen.prerecorded.v('1').transcribe_file(payload, options)



    # amend_url = get_safe_url(f"recordings/{filename}") #Presigned url from s3
    # response = deepgram.listen.prerecorded.v('1').transcribe_url({"url": amend_url}, options)

```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
hi @YupChutLuke 

Here is an example of increasing the timeout for Python if you have a large file or long audio stream.:
https://github.com/deepgram/deepgram-python-sdk/blob/main/examples/prerecorded/file/main.py#L51
